### PR TITLE
Handle platform on user's behalf in indexer_grpc_local.py

### PR DESCRIPTION
**Pull request by banool**
### Description
This PR makes a variety of improvements to indexer_grpc_local.py:
- The script now does a series of checks to ensure that the environment is correct, such as whether docker is present, running, has docker compose, etc. and whether grpcurl is installed.
- The script already tried to use the correct Redis image based on the machine, but this was actually incorrect for x86 Macs. It now handles this correctly.
- It also sets DOCKER_DEFAULT_PLATFORM based on the OS + CPU architecture so the user shouldn't have to worry about it.
- It is now possible to use this script on Linux where you can only use Docker as root by setting the `--run-docker-as-root` flag.
- We use `docker compose` rather than `docker-compose`.
- I remove the `timeout` dependency and handle it in Python instead. As part of this I simplify the run functions. I'm pretty sure they're correct but let's see how it looks in CI.
- Some of the argument parsing has been simplified.

I've also updated the docs:
- Updated the prereqs section.
- Rearranged some of the debugging stuff.
- Added more guidance for making sure the script works on ARM MacOS.

### Test Plan
I tested this manually on both ARM MacOS and x86 Linux.

Unit tests:
```
$ poetry run python -m unittest indexer_grpc_local_test.py
.
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK
```

### CODE REVIEW

**Comment by banool** [comment]
Addresses https://github.com/aptos-labs/aptos-core/issues/9878.

**Comment by banool** [comment]
Enabling auto merge to get some of the tests to run.

**Review by rustielin — APPROVED** [review]
LGTM! Grpc integration and e2e tests look good 

Looks a lot cleaner now -- thanks also for updating the docs and utils

File: testsuite/indexer_grpc_local.py [hunk 1]
```diff
@@ -172,106 +182,175 @@ def wait_for_indexer_grpc_progress(shell: Shell, client: HttpClient) -> None:
...
-            res.exit_code == 124
-        ):  # timeout exits with 124 if it reaches the end of the timeout
+        try:
+            res = context.shell.run(
+                [
+                    GRPCURL_PATH,
+                    "-max-msg-sz",
+                    "10000000",
+                    "-d",
+                    '{ "starting_version": 0 }',
```
Comment (larry-aptos) [inline testsuite/indexer_grpc_local.py:192]: optional: currently data service supports finite stream, i.e., `transactions_count`. The TimeoutExpired exception may indicate other issues if the finite stream is expected to end already.

**Review by larry-aptos — APPROVED** [review]

File: testsuite/indexer_grpc_local.py (hunk 1 above)
Comment (banool) [reply testsuite/indexer_grpc_local.py:192]: Okay good to know. I'll leave that as an optional improvement for now.

**Pull request merged**